tests/test_device_hardware.py

```
@@ -0,0 +1,693 @@
+    def test_very_long_device_pattern(self):
+        """Test handling of very long device patterns."""
+        long_pattern = "a" * 1000
+        manager = DeviceHardwareManager()
+
+        with patch.object(manager, "get_device_patterns", return_value={long_pattern}):
+            assert manager.is_device_pattern(long_pattern) is True
+            assert manager.is_device_pattern(long_pattern[:10]) is False
+
+    def test_unicode_device_patterns(self):
+        """Test handling of unicode device patterns."""
+        unicode_patterns = {"设备1", "девайс2", "デバイス3"}
```

Ruff (0.14.1)

String contains ambiguous `е` (CYRILLIC SMALL LETTER IE). Did you mean `e` (LATIN SMALL LETTER E)?

(RUF001)